### DISCUSSION TITLE
How to replace nginx certificate with a script

### DISCUSSION THREAD

**Discussion by CharlieMK** [Security-Onion-Solutions/securityonion#11105] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

tap

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

In 2.3 I simply moved a new cert and key file to /etc/pki. 

However, in 2.4.10 these files get overwritten, even if 
- I set replace_cert to true
- I replace the contents of ssl.key and ssl.cert in the nginx pki directory 

I can’t seem to find a pillar value corresponding to the grid values either. 

Any ideas? 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by EddieN17** [comment, marked as answer]
Place the cert and key in `/opt/so/saltstack/local/salt/nginx/ssl/ssl.(crt/key)`

or (I believe the recommended method in 2.4) use the web console, administration -> configuration, turn on advance settings, then go to nginx -> ssl and copy and paste your cert there.

**Comment by dougburks** [comment]
I've updated the docs:
https://docs.securityonion.net/en/2.4/nginx.html#replacing-default-cert
